### PR TITLE
fix(index): report 500 from received from backend as 502

### LINKED ISSUE #277: 500 from backend should be reported as 502
`helix-static` propagates status code `500` received from the backend as `500`. `502` would be more appropriate.

### PR DESCRIPTION
**Pull request by stefan-guggisberg**
fix #277

### CODE REVIEW

**Comment by stefan-guggisberg** [comment]
> Suggestion: Maybe it would be cleaner to implement this without mutating `statusCode` and just `error(502)` out of the branch.

I considered it. But then we'd miss the `log` statement with the `message`. WDYT?

**Comment by trieloff** [comment]
In the end we would want to have a custom path for each common status code, so missing the `log.error` below would be a good thing, because we could have a `log.warn` in the conditional. (I'd argue that only unhandled errors warrant a `log.error`.

File: src/index.js [hunk 1]
```diff
@@ -382,14 +382,17 @@ function deliverPlain(owner, repo, ref, entry, root, esi = false, branch, github
...
     }
-    if (rqerror.statusCode === 404 || rqerror.statusCode === '404') {
-      return error(entry, rqerror.statusCode);
+    const { statusCode, message, response } = rqerror;
+    if (statusCode === 404) {
+      return error(entry, statusCode);
     }
-    log.error('error while fetching content', rqerror.message);
+    if (statusCode === 500) {
+      return error(message, 502); // bad gateway
```
Comment (trieloff) [inline src/index.js:390]:
```suggestion
      log.warn('error 500 from backend: ' + message);
      return error(message, 502); // bad gateway
```

File: src/index.js [hunk 2]
```diff
@@ -382,14 +382,17 @@ function deliverPlain(owner, repo, ref, entry, root, esi = false, branch, github
...
+    }
+    log.error('error while fetching content', message);
```
Comment (trieloff) [inline src/index.js:392]:
```suggestion
    log.error('unknown error while fetching content', message);
```

**Pull request merged**